deployment/src/main/java/com/datastax/oss/quarkus/deployment/internal/CassandraClientProcessor.java

```diff
@@ -64,73 +64,90 @@ FeatureBuildItem feature() {
 
   @BuildStep
   List<ReflectiveClassBuildItem> registerGraphForReflection() {
+    ReflectiveClassBuildItem.Builder builder =
+        ReflectiveClassBuildItem.builder(
+                "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal")
+            .constructors(true)
+            .methods(true)
+            .fields(true);
```

Comment:
@msmygit (and @adutra if you have the cycles) this is the part of the upgrade I'm most concerned about.  I was seeing a lot of warnings about the deprecated constructors of ReflectiveClassBuildItem so it seemed worthwhile to shift over to the new builder usage.  Unfortunately I'm not completely paranoid that I might've accidentally dropped a class in translation so if either (or both) of you have cycles to give this a good once-over and make sure all the expected classes are still covered I'd greatly appreciate it.

Reply:
LGTM. Let's also get @adutra's eyes on it. 👍🏼

Reply:
Why do we need `.fields(true)` here and not elsewhere? The original code had `true, true` for all items in this method.

Reply:
Good eye @adutra!  The (boolean, boolean, String) [constructor](https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L66-L69) is for (methods, fields, classes).  So we _do_ need `.fields(true)` here... but `.constructors(true)` is not required.

I'll remove it.

Reply:
Done

Reply:
@absurdfarce I _believe_ `.constructors(true)` is indeed required because the other constructors `(methods,fields,classes)` is actually setting the `constructors` to `true` anyways [here](https://github.com/quarkusio/quarkus/blob/3.0.0.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L84-L96). WDYT?